players/group_3.py

```diff
@@ -49,15 +57,33 @@ def init_cuts(self, current_percept) -> Tuple[List[List[int]]]:
         ([[y1, y1], [y2, y2]], [[x1, x1], [x2, x2]])          # i.e., (HORIZONTAL_CUTS, VERTICAL_CUTS)
         """
         cake_len, cake_width = current_percept.cake_len, current_percept.cake_width
-        row_count, col_count = 2, 4
-        row_height, col_width = cake_len / row_count, cake_width / col_count
-
-        horizontal_cuts = [[i * row_height, i * row_height] for i in range(1, row_count)]
-        vertical_cuts = [[i * col_width, i * col_width] for i in range(1, col_count)]
 
-        print(f"{cake_len=}, {cake_width=}")
-        print(f"init_{horizontal_cuts=}, {vertical_cuts=}")
-        return horizontal_cuts, vertical_cuts
+        working_pairs = []
+        extra_pieces = 0
+        extra_area = cake_len * cake_width * 0.05
+        while not working_pairs:
+            mod_requests = current_percept.requests
+            mod_requests += [extra_area / extra_pieces for _ in range(extra_pieces)]
+            factor_pairs = self.find_factor_pairs(len(mod_requests))
+
+            for factor_pair in factor_pairs:
+                num_rows, num_cols = factor_pair
+                row_height, col_width = cake_len / num_rows, cake_width / num_cols
+                fits_plate = np.sqrt(row_height**2 + col_width**2) <= 25
```

Comment:
this made sense but then it got me thinking: is there any meaning to this if this algo actually just initializing the cuts and the subsequent descent/determination step can shift the cuts such that this condition might not hold?

Reply:
that would be another constraint within the actual algorithm, making sure that the pieces actually fit on the plates while altering the cuts

Reply:
~i was just wondering if there is a need for say like "1 very small row and 1 big row" for optimal solution -> total 2 rows, this might be ruled out if we divide the lengths equally and check whether each will fit on plate? e.g. length of whole cake = 25 and optimal solution cuts it to 1cm and 24cm rows but in this func it might be ...~ (this is where i realized i'm speaking nonsense) 

^ nonsense comment that doesn't make sense because current func should output all cases now that i think about it. still posting it cuz a similar line of reasoning may be relevant in later step lol